pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -570,23 +571,102 @@ type CloudConfig struct {
 		//issue body.
 		DisableStrictZoneCheck bool
 	}
+	// [ServiceOverride "1"]
+	//  Service = s3
+	//  Region = region1
+	//  URL = https://s3.foo.bar
+	//  SigningRegion = signing_region
+	//
+	//  [ServiceOverride "2"]
+	//     Service = ec2
+	//     Region = region2
+	//     URL = https://ec2.foo.bar
+	//     SigningRegion = signing_region
+	ServiceOverride map[string]*struct {
+		Service       string
+		Region        string
+		URL           string
+		SigningRegion string
```

Comment:
Can you add `SigningMethod` and pass that to [`endpoints.ResolvedEndpoint.SigningMethod`](https://docs.aws.amazon.com/sdk-for-go/api/aws/endpoints/#ResolvedEndpoint)?

Reply:
Can this be optional? In our testing, we did not set this and everything worked. Does it default to something sensible in us-east-1? That is where we were testing. In air gapped, maybe the default will be wrong?

Reply:
You're right that it will work without being set, as an empty value defaults to `v4`. This will help AWS with testing

Reply:
so use it as is, but do not validate it (check for empty)?